doc/rpc.rst

```diff
@@ -0,0 +1,54 @@
+.. _rpc:
+
+********
+Dune RPC
+********
+
+Starting from dune 3.0, dune's watch mode also runs an RPC server. This is a
+general mechanism introduced to integrate with various dune functionality. Some
+use cases we have in mind are:
+
+- Text editors that would like to receive dune's error reporting
+- In the future, custom actions that need to communicate with dune
+
+There's no fixed scope for RPC, and we encourage users to submit requests to
+cover more functionality.
+
+The purpose of this documentation is to explain how RPC works, and how to
+connect to it as a client. More concrete information such as what requests are
+available is in Dune_rpc_
+
+`dune-rpc` library
+==================
+
+We provide a client library ``dune-rpc`` to make it easy to write clients. The
+library has a versioned interface and we guarantee to maintain its stability.
+The library documentation describes the guarantees in more detail.
+
+RPC clients written with this library are guaranteed to work for all versions of
+the rpc server greater than it. In other words, a client using version ``X``
+will work with all servers ``>= X``. However, it will not work with any servers
```

Comment:
I think it can be clarified that in that case the server is dune and so `X` is a version of the dune binary (and I suppose that it's all unrelated from the dune-lang version)

Reply:
Hmm, you mind pushing a fix to my PR here? I'm not quite sure which PR needs clarification.

Reply:
Feel free to propose a fix in a PR